### PR TITLE
SortingRendererClass work

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/w3d/renderer/dx8wrapper.h [hunk 1]
```diff
@@ -781,3 +786,60 @@ inline void DX8Wrapper::Set_Index_Buffer_Index_Offset(unsigned offset)
...
+    }
+
+    for (int i = 0; i < VERTEX_BUFFERS; i++) {
+        if (s_renderState.vertex_buffers[i]) {
+            s_renderState.vertex_buffers[i]->Release_Engine_Ref();
+        }
+    }
+
+    s_renderState = state;
+    s_renderStateChanged = 0xffffffff;
```
Comment (duncanspumpkin) [inline src/w3d/renderer/dx8wrapper.h:808]: L787 used this like a set of flags, one of them looks wrong unsure which.

Comment (tomsons26) [reply src/w3d/renderer/dx8wrapper.h:808]: Checked pseudo, it's correct, i guess it's a catch all for all states have changed

**Comment by tomsons26** [comment]
Break up the lines so its not one monolith, new line after multiline, new line when context/operation changes( it sets polygon variables, then sets vertex variables - a context change), space out asserts

**Comment by tomsons26** [comment]
Works, seems good otherwise at least as far as i can tell

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**